src/overlays/actors/ovl_Bg_Dkjail_Ivy/z_bg_dkjail_ivy.c

```diff
@@ -28,47 +36,176 @@ const ActorInit Bg_Dkjail_Ivy_InitVars = {
+        BUMP_ON,
+        OCELEM_NONE,
+    },
     { 20, 80, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80ADE98C[] = {
+static Gfx* sLeafDlists[] = { gKakeraLeafMiddle, gKakeraLeafTip };
+
+static s16 sLeafScales[] = { 110, 80, 60, 40 };
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 200, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
-
-extern ColliderCylinderInit D_80ADE950;
-extern InitChainEntry D_80ADE98C[];
-
-extern UNK_TYPE D_06000080;
-extern UNK_TYPE D_060011A8;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE230.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/BgDkjailIvy_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/BgDkjailIvy_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE6AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE6C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE734.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE748.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE7E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE7F4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/BgDkjailIvy_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/BgDkjailIvy_Draw.s")
+void BgDkjailIvy_IvyCutEffects(BgDkjailIvy* this, PlayState* play) {
+    f32 phi_fs0;
+    s32 i;
+    Vec3f spD4;
+    Vec3f spC8;
+    Vec3f spBC;
+    Vec3f spB0;
+    s16 i2;
+
+    Matrix_RotateYS(this->dyna.actor.home.rot.y, MTXMODE_NEW);
+
+    for (i = 0, i2 = 0; i < 50; i++, i2 += 0x4e20) {
+        spD4.y = (i * (90.0f / 49.0f)) + 2.0f;
+        if (spD4.y < 45.0f) {
+            phi_fs0 = ((1.0f / 180.0f) * spD4.y) + 0.75f;
+        } else if (spD4.y < 90.0f) {
+            phi_fs0 = ((-1.0f / 180.0f) * spD4.y) + 1.25f;
+        } else {
+            phi_fs0 = ((-1.0f / 30.0f) * spD4.y) + 3.75f;
+        }
+
+        spD4.x = Math_SinS(i2) * 40.0f * phi_fs0;
+        spD4.z = (Rand_ZeroOne() * 6.0f) - 3.0f;
+
+        Matrix_MultVec3f(&spD4, &spC8);
+
+        spBC.x = (Rand_ZeroOne() - 0.5f) + (spC8.x * 0.075f);
+        spBC.y = 2.0f * Rand_ZeroOne();
+        spBC.z = (Rand_ZeroOne() - 0.5f) + (spC8.z * 0.075f);
+
+        spC8.x += this->dyna.actor.world.pos.x;
+        spC8.y += this->dyna.actor.world.pos.y;
+        spC8.z += this->dyna.actor.world.pos.z;
+
+        EffectSsKakera_Spawn(play, &spC8, &spBC, &spC8, -0x82, 0x40, 0x28, 0, 0, sLeafScales[i & 3], 0, 0, 44, -1,
+                             GAMEPLAY_KEEP, sLeafDlists[(s32)Rand_Next() > 0]);
```

Comment:
Can name position/velocity/accel after the arguments. I think the -0x82 is gravity so that should be dec too.